staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -709,12 +705,12 @@ func (s *store) getState(getResp *clientv3.GetResponse, key string, v reflect.Va
 			return nil, err
 		}
 	} else {
+		state.rev = getResp.Kvs[0].ModRevision
+		state.meta.ResourceVersion = uint64(state.rev)
 		data, stale, err := s.transformer.TransformFromStorage(getResp.Kvs[0].Value, authenticatedDataString(key))
 		if err != nil {
-			return nil, storage.NewInternalError(err.Error())
+			return state, storage.NewTransformerError(err.Error())
```

Comment:
where are we using the returned data even in the presence of an error?

Reply:
Did you mean "why"?

The state only contains `state.rev`, which comes from `getResp.Kvs[0].ModRevision`. The transformer error only indicates that the `getResp.Kvs[0].Value` is corrupted.

Reply:
I meant where. Wanted to see how we were checking the value was usable/non-nil. Documentation should be explicit about what parts of the returned object are valid in the presence of an error

Reply:
Right. Added documentation in commit https://github.com/kubernetes/kubernetes/pull/76346/commits/4a5699ad2cb1cea337d5b60b7251787ed65fa728.

The `state.rev` is used in https://github.com/kubernetes/kubernetes/pull/76346/files#diff-0f11327dd3daafdfe8e326e895616f66R236 in case of TranformError is returned.